plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/PluginWithIntegrationTestsHook.java

```diff
@@ -1,24 +1,36 @@
 package org.jenkins.tools.test.hook;
 
-import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
-
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
+
 /**
- * Workaround for those plugins with integration tests since they need execute the failsafe:integration-test goal before execution.
+ * Hook for plugins with integration tests that need to be executed 
  */
 public abstract class PluginWithIntegrationTestsHook extends PluginCompatTesterHookBeforeExecution {
 
+    /** Inform about goals to execute integration tests */
+    abstract public Collection<String> getGoals();
```

Comment:
Are we sure these are not being used outside? Also having nullability annotations would be nice I think.

Reply:
Yes, AFAIK seems only used here (see also: https://github.com/search?q=PluginWithIntegrationTestsHook&type=code) Added NonNull annotation too here https://github.com/jenkinsci/plugin-compat-tester/pull/258/commits/b6922615e6d8e4752ef2a0e37abaa1d818ccdb95